### DISCUSSION TITLE
If some form items are dynamically generated based on data and can't be registered with default values at the beginning, can they only be registered with "reset"?

### DISCUSSION THREAD

**Discussion by shervinchen** [react-hook-form/react-hook-form#9790] [Q&A]
The initial data looks like this:
```tsx
const form = useForm({
  defaultValues: {
    name: 'tom',
    age: '22',
  },
  resolver: yupResolver(yup.object().shape({
    name: yup.string().required('Please put in name'),
    age: yup.string().required('Please put in age'),
  })),
})
```
But there are still some form items that are generated through backend data, so currently I am using the `reset` API to reset the data
```tsx
const commonForms = {
  name: form.getValues('name'),
  age: form.getValues('age')
}
const formValues = {}
data.forEach(item => {
  formValues[item.id] = item.value
})
form.reset({ ...commonForm, ...formValues })
```
I found that the `reset` API does not reset specific form items individually, but resets the values of all forms, so I manually filled those fields that I don't want to be reset again, which I think is a tedious approach.

I only want to reset those dynamically generated form items without affecting the default form items, how can I do that?
(By the way, I found that the `resetField` API can reset individual fields, but these fields must be registered at the beginning. And in my case, those dynamically generated form fields cannot be registered at the beginning because the key name has not been determined yet)

**Comment by bluebill1049** [comment]
Is the keep option is what you after?

```tsx
reset(newData, {
  keepDefaultValues: true,
  keepValues: true,
})
```
